src/compiler/checker.ts

```diff
@@ -20770,6 +20770,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             const sourceType = i === restIndex ? getRestOrAnyTypeAtPosition(source, i) : tryGetTypeAtPosition(source, i);
             const targetType = i === restIndex ? getRestOrAnyTypeAtPosition(target, i) : tryGetTypeAtPosition(target, i);
             if (sourceType && targetType) {

```

Comment:
Change to `sourceType && targetType && sourceType !== targetType` and get rid of the conditional `continue`.

Reply:
done :) thanks for pointing out